src/creatwth.cpp

```diff
@@ -557,46 +587,30 @@ BOOL WINAPI DetourUpdateProcessWithDll(_In_ HANDLE hProcess,
 
         if ((inh.FileHeader.Characteristics & IMAGE_FILE_DLL) == 0) {
             hModule = hLast;
+            if (inh.OptionalHeader.Magic == IMAGE_NT_OPTIONAL_HDR32_MAGIC
+                && inh.FileHeader.Machine != 0) {
+
+                bHas32BitExe = TRUE;
+            }
             DETOUR_TRACE(("%p  Found EXE\n", hLast));
         }
+        else {
+            if (inh.OptionalHeader.Magic == IMAGE_NT_OPTIONAL_HDR64_MAGIC
+                && inh.FileHeader.Machine != 0) {
+
+                bHas64BitDll = TRUE;
```

Comment:
Value is never read?

Reply:
this above code is just a revert of 81e6a5f79645b6fe385a2b72ee728962049dee6b, It has always existed in the previous version. It should have been written by one of your microsoft technical experts. Keeping it is only as a respect for the original author. Keeping it will not affect it. Maybe it will be used in the future.

Reply:
![image](https://user-images.githubusercontent.com/20564099/92407808-3ff5ee80-f16e-11ea-9add-e6fae3f35d4d.png)
from the IDA decompiled code, we can known that, my guess is correct, the smart VC compiler can really optimize out the code that is not used in the function.